build/scripts/base_image.sh

```diff
@@ -32,7 +32,7 @@ fi
 
 # build docker image
 build_image()  {
-    docker build -t "${DOCKER_IMAGE}" \
+    DOCKER_BUILDKIT=1 docker build --progress=plain -t "${DOCKER_IMAGE}" \
```

Comment:
is buildkit necessary, for cache or what?

Reply:
@Effet buildkit is required to support dockerfile:1.2 syntax. Plain progress is better for devs to read logs.

Reply:
What if our docker daemon version not support buildkit?

Reply:
Just make sure the docker daemon version is later than 18.09, and it's 19.03.5 on cloud.

Reply:
```
2021-07-01 15:51:36+ docker version
2021-07-01 15:51:37Client:
2021-07-01 15:51:37 Version:           20.10.3
2021-07-01 15:51:37 API version:       1.40
2021-07-01 15:51:37 Go version:        go1.13.15
2021-07-01 15:51:37 Git commit:        48d30b5b32e99c932b4ea3edca74353feddd83ff
2021-07-01 15:51:37 Built:             Wed Feb 17 17:47:43 2021
2021-07-01 15:51:37 OS/Arch:           linux/amd64
2021-07-01 15:51:37 Context:           default
2021-07-01 15:51:37 Experimental:      true
2021-07-01 15:51:37
2021-07-01 15:51:37Server: Docker Engine - Community
2021-07-01 15:51:37 Engine:
2021-07-01 15:51:37  Version:          19.03.5
2021-07-01 15:51:37  API version:      1.40 (minimum version 1.12)
2021-07-01 15:51:37  Go version:       go1.12.12
2021-07-01 15:51:37  Git commit:       99c5edce
2021-07-01 15:51:37  Built:            Thu Jan  9 08:26:30 2020
2021-07-01 15:51:37  OS/Arch:          linux/amd64
2021-07-01 15:51:37  Experimental:     false
2021-07-01 15:51:37 containerd:
2021-07-01 15:51:37  Version:          1.2.10
2021-07-01 15:51:37  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
2021-07-01 15:51:37 runc:
2021-07-01 15:51:37  Version:          1.0.0-rc8+dev
2021-07-01 15:51:37  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
2021-07-01 15:51:37 docker-init:
2021-07-01 15:51:37  Version:          0.18.0
2021-07-01 15:51:37  GitCommit:        fec3683
2021-07-01 15:51:37
```

Reply:
I also checked yesterday, and it's ok for erda itself CI.
For customers, we will use another buildkit plan.